drivers/azure/azureutil/customimage.go

```diff
@@ -0,0 +1,116 @@
+	resourceGroupRegex = regexp.MustCompile(`[rR]esource[gG]roups/[-\w\._\(\)]+`)
+	// The name must begin with a letter or number, end with a letter, number or underscore, and may contain only letters, numbers, underscores, periods, or hyphens
+	imageRegex = regexp.MustCompile(`[iI]mages/[A-Za-z][A-Za-z0-9-_]{1,61}[A-Za-z0-9_]`)
+	// Shared image gallery name shouldn't contain hyphens, may contain letters, numbers and allow underscores and periods in the middle.
+	galleryRegex = regexp.MustCompile(`[gG]alleries/[A-Za-z0-9][A-Za-z0-9_.]{1,61}[A-Za-z0-9]`)
+	// The image version name should follow Major(int).Minor(int).Patch(int) format
+	versionRegex = regexp.MustCompile(`[vV]ersions/[0-9]*\.[0-9]*\.[0-9]*`)
+
+	// NOTE(aiyengar2): these value is optional as it isn't required for the get request
+	subscriptionsFormat = "subscriptions/%s"
+	// Regex: https://docs.microsoft.com/bs-latn-ba/rest/api/securitycenter/locations/get
+	subscriptionRegex = regexp.MustCompile(`[sS]ubscriptions/[0-9A-Fa-f]{8}-([0-9A-Fa-f]{4}-){3}[0-9A-Fa-f]{12}`)
+)
+
+const (
+	invalidImageErrorString = "Failure parsing image %q. " +
+		"If you intended to provide a custom image name, we require the ARM resource identifier for that custom image. See " +
+		"https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.management.batch.models.imagereference.id?view=azure-dotnet#Microsoft_Azure_Management_Batch_Models_ImageReference_Id " +
+		"for more information on how to supply a custom image's ARM resource identifier. "
+	versionNotSuppliedWarnString = "Version is missing; please supply a specific image version for the gallery image definition specified"
+)
+
+type customImage struct{ resourceGroup, imageName, galleryName, versionID, subscriptionID string }
+
+// Parses a custom image based on https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.management.batch.models.imagereference.id?view=azure-dotnet#Microsoft_Azure_Management_Batch_Models_ImageReference_Id
+func parseCustomImageReference(image string) (*customImage, error) {
+	// NOTE(aiyengar2): the custom image ID provided must adhere to the ARM resource identifier format as we
+	// do not have information about the resourceGroup that contains the requested custom image.
+	// customImageFormat = "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Compute/images/%s"
+	numExpectedRequiredValues := 2
+	if strings.Contains(image, "galleries/") {
+		numExpectedRequiredValues = 4
+	}
+	l := [5]string{
+		resourceGroupRegex.FindString(image),
+		imageRegex.FindString(image),
+		galleryRegex.FindString(image),
+		versionRegex.FindString(image),
+		subscriptionRegex.FindString(image),
+	}
+	ltrimmed := [5]string{}
+	for i, value := range l {
+		prefixRemoved := strings.Split(value, "/")
+		if len(prefixRemoved) == 2 {
+			ltrimmed[i] = prefixRemoved[1]
+		} else if i < numExpectedRequiredValues {
+			// failed to parse required values
+			if i == 3 {
+				log.Warn(versionNotSuppliedWarnString)
```

Comment:
If this is required when using gallery, can we make it part of the error? The error should contain enough to root cause the issue, and while the generic error probably covers it, we do know what the error is but now its only logged in a different level and on a separate line. If it's not required/breaking then this can be the additional warning

Reply:
Good idea! Should be fixed in the latest commit